app/controllers/dashboards_controller.rb

```diff
@@ -56,10 +56,11 @@ def followers
 
   def analytics
     @user_or_org = if params[:org_id]
-                     Organization.find_by(id: params[:org_id])
+                     Organization.find(params[:org_id])
```

Comment:
Question and not blocking: I'm curious if the `find_by` was intentional given that it _does not_ raise the `ActiveRecord::RecordNotFound` error if the record is not found?

Reply:
This is a really good catch on a subtle change.  Prior to this commit, if we didn't find the org, we had `@user_or_org == nil`.  Which didn't matter, because later on we'd check `not_authorized if params[:org_id] && !@user.org_admin?(params[:org_id] || @user.any_admin?)` (see https://github.com/forem/forem/blob/7f9681465eae6aaaf2d11721618008a55ee354b7/app/controllers/dashboards_controller.rb#L10).  That code has a logic bug based on parentheses, which I have only just seen and has existed for awhile.  So

That part would raise `not_authorized` because the user was not an org admin for the given ID OR true/false (note the parentheses).

The proposed chanbge  short-circuits the authorization antics by raising `ActiveRecord::NotFoundError` if the organization doesn't exist.

From the outside looking in, this change does provide a crease to say "Oh, instead of Forbidden we get Not Found".

Reply:
ah, ok. thanks for tying all that together 👍